File: src/main/java/org/axostudio/axohologram/animation/AnimationFrameCache.java
package org.axostudio.axohologram.animation;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;
import java.util.function.Supplier;

public final class AnimationFrameCache {

    private final Map<String, String> textFrames = new ConcurrentHashMap<>();

    public String getTextFrame(String key, Supplier<String> renderer) {
        return textFrames.computeIfAbsent(key, ignored -> renderer.get());
    }

    public void clear() {
        textFrames.clear();
    }
}

File: src/main/kotlin/org/axostudio/axohologram/feature/animation/runtime/AnimationFrameCache.kt
package org.axostudio.axohologram.feature.animation.runtime

import org.axostudio.axohologram.feature.animation.display.DisplayAnimationFrame
import java.util.concurrent.ConcurrentHashMap

class AnimationFrameCache {

    private val cache = ConcurrentHashMap<String, List<DisplayAnimationFrame>>()

    fun getFrames(animationName: String): List<DisplayAnimationFrame>? = cache[animationName.lowercase()]

    fun putFrames(animationName: String, frames: List<DisplayAnimationFrame>) {
        cache[animationName.lowercase()] = frames
    }

    fun clear() {
        cache.clear()
    }
}
